CHANGES.rst

```diff
@@ -1,3 +1,447 @@
+
+- ``clone`` has been enhanced to allow for flat cosmologies to clone on the
+  equivalent non-flat cosmology. [#13099]
+
+- ``cosmology`` file I/O uses the Unified Table I/O interface, which has added
+  support for reading and writing file paths of the form ``~/file.ecsv`` or
+  ``~<username>/file.ecsv``, referring to the home directory of the current user
+  or the specified user, respectively. [#13129]
+
+astropy.io.ascii
+^^^^^^^^^^^^^^^^
+
+- Simplify the way that the ``converters`` argument of ``io.ascii.read`` is
+  provided. Previously this required wrapping each data type as the tuple returned
+  by the ``io.ascii.convert_numpy()`` function and ensuring that the value is a
+  ``list``. With this update you can write ``converters={'col1': bool}`` to force
+  conversion as a ``bool`` instead of the previous syntax ``converters={'col1':
+  [io.ascii.convert_numpy(bool)]}``. Note that this update is back-compatible with
+  the old behavior. [#13073]
+
+- Added support in ``io.ascii`` for reading and writing file paths of the form
+  ``~/file.csv`` or ``~<username>/file.csv``, referring to the home directory of
+  the current user or the specified user, respectively. [#13130]
+
+astropy.io.fits
+^^^^^^^^^^^^^^^
+
+- Add option ``unit_parse_strict`` to ``astropy.io.fits.connect.read_table_fits``
+  to enable warnings or errors about invalid FITS units when using ``astropy.table.Table.read``.
+  The default for this new option is ``"warn"``, which means warnings are now raised for
+  columns with invalid units. [#11843]
+
+- Changes default FITS behavior to use buffered I/O
+  rather than unbuffered I/O for performance reasons. [#12081]
+
+- ``astropy.io.fits.Header`` now has a method to calculate the size
+  (in bytes) of the data portion (with or without padding) following
+  that header. [#12110]
+
+astropy.io.misc
+^^^^^^^^^^^^^^^
+
+- Allow serialization of model unit equivalencies. [#10198]
+
+- Built-in Cosmology subclasses can now be converted to/from YAML with the
+  functions ``dump`` and ``load`` in ``astropy.io.misc.yaml``. [#12279]
+
+- Add asdf support for ``Cosine1D``, ``Tangent1D``, ``ArcSine1D``,
+  ``ArcCosine1D``, and ``ArcTangent1D`` models. [#12895]
+
+- Add asdf support for ``Spline1D`` models. [#12897]
+
+astropy.io.registry
+^^^^^^^^^^^^^^^^^^^
+
+- Added support to the Unified Table I/O interface for reading and writing file
+  paths of the form ``~/file.csv`` or ``~<username>/file.csv``, referring to the
+  home directory of the current user or the specified user, respectively. [#13129]
+
+astropy.modeling
+^^^^^^^^^^^^^^^^
+
+- Add new fitters based on ``scipy.optimize.least_squares`` method of non-linear
+  least-squares optimization:
+
+      - ``TRFLSQFitter`` using the Trust Region Reflective algorithm.
+      - ``LMLSQFitter`` using the Levenberg-Marquardt algorithm (implemented by ``scipy.optimize.least_squares``).
+      - ``DogBoxLSQFitter`` using the dogleg algorithm. [#12051]
```

Comment:
I don't know where one should put the PR number in a list like this. Do we allow sub-list in change log?

Reply:
A sub-list seems more readable in this case (even if we should to try to keep changelog entries short, in cases like this I think it's better to take a bit of space to explain the changes). Not sure for the PR number though, I would probably put it on the first line.

Reply:
First line sounds good to me. Thanks!